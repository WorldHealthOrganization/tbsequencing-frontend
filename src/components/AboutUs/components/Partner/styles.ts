import { css } from '@emotion/react';
import appColors from '../../../../styles/colors';

export const container = css({
  width: 200,
  display: 'flex',
  flexWrap: 'wrap',
  flexDirection: 'column',
  alignItems: 'center',
});

export const title = css({
  marginBottom: 8,
  textAlign: 'center',
});

export const description = css({
  fontSize: 16,
  lineHeight: '24px',
  textAlign: 'center',
});

export const link = css({
  color: appColors.neutral.black,
  textDecoration: 'none',
  ':hover': {
    textDecoration: 'underline',
    textDecorationColor: appColors.neutral.black,
  },
  display: 'flex',
});
