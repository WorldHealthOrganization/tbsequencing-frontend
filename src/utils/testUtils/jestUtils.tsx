import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
