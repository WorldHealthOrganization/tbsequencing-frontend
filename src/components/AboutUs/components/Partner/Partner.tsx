/** @jsxImportSource @emotion/react */
import React, { ReactElement } from 'react';
import { SerializedStyles } from '@emotion/react';
import * as styles from './styles';
import H3 from '../../../typography/H3';
import PrimaryText from '../../../typography/PrimaryText';

interface Props {
  image: ReactElement;
  title: string;
  webSite: string;
  description: string;
  marginStyle?: SerializedStyles;
}

export const Partner = ({
  image, description, title, webSite, marginStyle,
}: Props) => (
  <div css={[styles.container, marginStyle]}>
    {image}
    <H3 style={styles.title}><a css={styles.link} href={webSite}>{title}</a></H3>
    <PrimaryText style={styles.description} css={styles.description}>{description}</PrimaryText>
  </div>
);

export default Partner;
