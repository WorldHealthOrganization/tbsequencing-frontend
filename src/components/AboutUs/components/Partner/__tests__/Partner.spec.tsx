import React from 'react';
import { render, getTreeJSONWithRedux } from '../../../../../utils/testUtils/jestUtils';
import Partner from '../Partner';

describe('Partner component test', () => {
  test('renders Partner component without error', () => {
    render(
      <Partner />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <Partner />,
    );

    expect(tree).toMatchSnapshot();
  });
});
