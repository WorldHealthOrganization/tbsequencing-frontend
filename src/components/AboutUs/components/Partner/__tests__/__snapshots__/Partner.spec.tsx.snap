// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partner component test Should render correctly 1`] = `
<div
  className="css-eqjams-Partner"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-kxf6at-MuiTypography-root-BaseText"
  >
    <a
      className="css-fe89j9-Partner"
    />
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1 css-nhq2dv-MuiTypography-root-BaseText"
  />
</div>
`;
